src/Features/CSharp/Portable/Diagnostics/Analyzers/CSharpUseImplicitTypingDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,148 @@
+            title: s_Title,
+            messageFormat: s_Message,
+            category: DiagnosticCategory.Style,
+            defaultSeverity: DiagnosticSeverity.Info,
+            isEnabledByDefault: true);
+
+        public CSharpUseImplicitTypingDiagnosticAnalyzer() : base(s_descriptorUseImplicitTyping)
+        {
+
+        }
+
+        protected override bool IsStylePreferred(SyntaxNode declarationStatement, SemanticModel semanticModel, OptionSet optionSet, CancellationToken cancellationToken)
+        {
+            var stylePreferences = GetCurrentTypingStylePreferences(optionSet);
+
+            var isTypeApparent = IsTypeApparentInDeclaration(declarationStatement, semanticModel, stylePreferences, cancellationToken);
+            var isIntrinsicType = IsIntrinsicType(declarationStatement);
+
+            return stylePreferences.HasFlag(TypingStyles.VarForIntrinsic) && isIntrinsicType
+                || stylePreferences.HasFlag(TypingStyles.VarWhereApparent) && isTypeApparent
+                || stylePreferences.HasFlag(TypingStyles.VarWherePossible) && !(isIntrinsicType || isTypeApparent);
+        }
+
+        protected override bool AnalyzeVariableDeclaration(TypeSyntax typeName, SemanticModel semanticModel, OptionSet optionSet, CancellationToken cancellationToken, out TextSpan issueSpan)
+        {
+            issueSpan = default(TextSpan);
+
+            // If it is already var, return.
+            if (typeName.IsTypeInferred(semanticModel))
+            {
+                return false;
+            }
+
+            var candidateReplacementNode = SyntaxFactory.IdentifierName("var")
+                                                .WithLeadingTrivia(typeName.GetLeadingTrivia())
+                                                .WithTrailingTrivia(typeName.GetTrailingTrivia());
+
+            var candidateIssueSpan = typeName.Span;
+
+            // If there exists a type named var, return.
+            var conflict = semanticModel.GetSpeculativeSymbolInfo(typeName.SpanStart, candidateReplacementNode, SpeculativeBindingOption.BindAsTypeOrNamespace).Symbol;
+            if (conflict != null && conflict.IsKind(SymbolKind.NamedType))
+            {
+                return false;
+            }
+
+            if (typeName.Parent.IsKind(SyntaxKind.VariableDeclaration) &&
+                typeName.Parent.Parent.IsKind(SyntaxKind.LocalDeclarationStatement, SyntaxKind.ForStatement, SyntaxKind.UsingStatement))
+            {
+                var variableDeclaration = (VariableDeclarationSyntax)typeName.Parent;
+
+                // implicitly typed variables cannot be constants.
+                var localDeclarationStatement = variableDeclaration.Parent as LocalDeclarationStatementSyntax;
+                if (localDeclarationStatement != null && localDeclarationStatement.IsConst)
+                {
+                    return false;
+                }
+
+                var variable = variableDeclaration.Variables.Single();
```

Comment:
(ShouldAnalyzeVariableDeclaration)[https://github.com/dotnet/roslyn/pull/8045/files#diff-38d5dc0c4ed785f53fde6c664f86c56dR252] filters out such declarations, as you could not use var on them.